delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageSaveFile.scala

```diff
@@ -1,76 +1,169 @@
 
+import akka.NotUsed
 import akka.actor.ActorSystem
 import akka.http.scaladsl.model.Uri.Path
-import akka.http.scaladsl.model.Uri.Path.Slash
 import akka.http.scaladsl.model.{BodyPartEntity, Uri}
-import akka.stream.alpakka.s3.scaladsl.S3
-import akka.stream.alpakka.s3.{S3Attributes, S3Exception}
-import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
 import cats.effect.IO
 import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.kernel.Logger
 import ch.epfl.bluebrain.nexus.delta.kernel.utils.UUIDF
 import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileAttributes.FileAttributesOrigin.Client
-import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileStorageMetadata
-import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.StoragesConfig.StorageTypeConfig
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.{Digest, FileStorageMetadata}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.DigestAlgorithm
 import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.Storage.S3Storage
 import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.SaveFile
-import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.SaveFile.{digestSink, intermediateFolders, sizeSink}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.SaveFile.intermediateFolders
 import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.StorageFileRejection.SaveFileRejection._
-import ch.epfl.bluebrain.nexus.delta.plugins.storage.utils.SinkUtils
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.s3.client.S3StorageClient
+import ch.epfl.bluebrain.nexus.delta.rdf.syntax.uriSyntax
+import ch.epfl.bluebrain.nexus.delta.sdk.stream.StreamConverter
+import eu.timepit.refined.refineMV
+import eu.timepit.refined.types.string.NonEmptyString
+import fs2.Stream
+import fs2.aws.s3.S3
+import fs2.aws.s3.S3.MultipartETagValidation
+import fs2.aws.s3.models.Models.{BucketName, ETag, FileKey, PartSizeMB}
+import software.amazon.awssdk.services.s3.model._
 
 import java.util.UUID
-import scala.concurrent.Future
 
-final class S3StorageSaveFile(storage: S3Storage, config: StorageTypeConfig)(implicit
+final class S3StorageSaveFile(s3StorageClient: S3StorageClient, storage: S3Storage)(implicit
     as: ActorSystem,
     uuidf: UUIDF
 ) extends SaveFile {
-  import as.dispatcher
-  private val fileAlreadyExistException = new IllegalArgumentException("Collision, file already exist")
 
-  override def apply(
+  private val client                  = s3StorageClient.underlyingClient
+  private val s3                      = S3.create(client)
+  private val multipartETagValidation = MultipartETagValidation.create[IO]
+  private val logger                  = Logger[S3StorageSaveFile]
+  private val partSizeMB: PartSizeMB  = refineMV(5)
```

Comment:
Could be configured, is that a recommended value ?
The `PutObject` operation is recommended up to 100MB (but can go further)